docs/application/native/guides/device/diagnostics.md

```diff
@@ -122,30 +125,81 @@ void crash_callback(diagnostics_ctx_h ctx, void *user_data)
 }
 ```
 
-## Diagnostic data request
+If `diagnostics_request_client_data()` returns successfully, the obtained data can be read by `diagnostics_data_read()`.
 
-Diagnostics API also can request data from other application or service, especially the diagnostic data that for various reasons cannot or must not be saved in the logs, such as binary or in custom format data, large volume data, reports that are time-consuming to create, and so on.
+Sometimes there is a need for getting data related to some specific diagnostics event. Ex. bugreport-service sends an event that a new bug report has been created and client wants to get the bug report's content after receiving the event. `diagnostics_get_data()` should be used in such case if application supports event-specific data request.
 
-To do so, call `diagnostics_request_client_data()`. The `client_id` variable must be the id of the process that supports Diagnostics API requests:
+# Current implementations
+
+## Bugreport service
+
+Bugreport service provides diagnostic interface to allow you to read the bug reports in case the process in the system fails. To do so, you must register the callback that will be called when the bug report is created:
```

Comment:
```suggestion
Bugreport service provides a diagnostic interface to allow you to read the bug reports in case the process in the system fails. To do so, you must register the callback `diagnostics_set_notification_cb`, it is called when the bug report is created:
```